app/src/services/kleros.ts

```diff
@@ -0,0 +1,133 @@
+        tokensViewContractAddress,
+        klerosTokensViewAbi,
+        provider,
+      )
+    }
+  }
+
+  /**
+   * Return the tokens from the KLEROS TCR already registered
+   * @returns {Promise<any[]>}
+   */
+  queryTokens = async () => {
+    const network = await this.provider.getNetwork()
+    const networkId = network.chainId
+
+    if (networkId !== networkIds.MAINNET) {
+      // Use mocked information from networks file
+      return getTokensByNetwork(networkId)
+    } else {
+      const filter = [
+        false, // Include absent tokens in result
+        true, // Include registered tokens in result.
+        false, // Include tokens with registration requests that are not disputed in result.
+        true, // Include tokens with clearing requests that are not disputed in result.
+        false, // Include disputed tokens with registration requests in result.
+        true, // Include disputed tokens with clearing requests in result.
+        false, // Include tokens submitted by the caller.
+        false, // Include tokens challenged by the caller.
+      ]
+
+      let hasMore = true
+      let addressesWithBadge: string[] = []
+
+      // Fetch addresses of tokens that have the badge.
+      // Since the contract returns fixed sized arrays, we must filter out unused items.
+      while (hasMore) {
+        const result = await this.badgeContract.queryAddresses(
+          ethers.constants.AddressZero,
+          1000,
+          filter,
+          true,
+        )
+        addressesWithBadge = addressesWithBadge.concat(
+          result[0].filter((address: string) => address !== ethers.constants.AddressZero),
+        )
+        hasMore = result.hasMore
+      }
+
+      // Fetch their submission IDs on the TCR.
+      const network = await this.provider.getNetwork()
+      const networkId = network.chainId
+      const klerosTCRAddress = getContractAddress(networkId, 'klerosTCR')
+      const submissionIDs = await this.tokensViewContract.getTokensIDsForAddresses(
+        klerosTCRAddress,
+        addressesWithBadge,
+      )
+
+      // With the token IDs, get the information and add it to the object.
+      const fetchedTokens = await this.tokensViewContract.getTokens(klerosTCRAddress, submissionIDs)
+      const tokens = fetchedTokens
+        .filter((tokenInfo: any) => tokenInfo[3] !== ethers.constants.AddressZero)
+        .map(
+          (token: any): Token => {
+            return {
+              symbol: token.ticker,
+              address: token.addr,
+              decimals: token.decimals.toNumber() || 18, // Token does not implement the 'decimals()' function, falling back to the default, 18 decimal places.
+            }
+          },
+        )
+
+      const basicOrder = ['dai', 'cdai', 'weth', 'usdc', 'owl', 'chai'].reverse()
+
+      return tokens.sort((tokenA: Token, tokenB: Token) => {
+        let sortBy: number
+
+        const indexOfB = basicOrder.indexOf(tokenB.symbol.toLowerCase())
```

Comment:
This is going to be extremely pedantic but I'm going to do it anyway: A should go before B.

Reply:
Done!